### DISCUSSION TITLE
Multithreading error (sqlite lock)

### DISCUSSION THREAD

**Discussion by Baimanight** [ROBelgium/MSNoise#350] [Q&A] [answered]
Recently，when I try to use '-t ', I always encounter that  'sqlite3.OperationalError) database is locked' . My version is 1.6.3. Is there any method ? Thanks very much ! By the way,  is MsNoise2.0 released？Or say it is just the development  version？

**Comment by ThomasLecocq** [comment, marked as answer]
Hi, indeed, as stated in the documentation, the usage of threading can lead to issues with sqlite, as sqlite is a non-parallel "write" database. If you use it, make sure to use the "-d 5" option (5 being the number of seconds to wait between threads).

msnoise 2 should be there soon, but "master" version is already very stable

**Reply by Baimanight** [reply]
Thanks for your response, unfortunately, it does not work on my computer, even just -t 2 -d 10
I also have difficulty in installation, the test did not pass. Is it because of ObsPy?
Maybe It is the master who can use the “master” version.

conda install -c conda-forge flask-admin flask-wtf markdown folium pymysql logbook pandas pytables pip xarray
pip install http://msnoise.org/master.zip
pip install https://github.com/obspy/obspy/archive/master.zip
![截图 2024-07-08 17-48-48](https://github.com/ROBelgium/MSNoise/assets/137660437/f09977ed-fdd2-442d-b30e-f033fc357bb1)
---------------+MODULES+-------------------

Required:
[X] setuptools: 70.1.1
[X] numpy: 1.26.4
[X] scipy: 1.14.0
[X] pandas: 2.2.2
[X] matplotlib: 3.9.1
[X] sqlalchemy: 1.4.52
[X] obspy: 0.0.0+archive
[X] click: 8.1.7
[X] pymysql: 1.4.6
[X] flask: 3.0.3
[X] flask_admin: 1.6.1
[X] markdown: 3.6
[X] wtforms: 3.1.2
[X] folium: 0.17.0
[X] jinja2: 3.1.4
[X] tables: 3.9.2 (pytables)
[X] xarray: 2024.6.0
[X] logbook: 1.7.0.post0

Only necessary if you plan to build the doc locally:
[ ] sphinx: not found
[ ] sphinx_bootstrap_theme: not found
[ ] sphinx_rtd_theme: not found
[ ] sphinx_gallery: not found
[ ] numpydoc: not found

Graphical Backends:
[ ] wx: not found
[ ] qt: not found
[ ] qt4: not found
[ ] qt5: not found
[ ] pyqt: not found
[ ] PyQt4: not found
[X] PyQt5: present (no version)
[ ] PySide: not found

Not required, just checking:
[X] json: 2.0.9
[X] psutil: 6.0.0
[ ] reportlab: not found
[ ] configobj: not found
[X] pkg_resources: present (no version)
[ ] paramiko: not found
[X] ctypes: 1.1.0
[X] pyparsing: 3.1.2
[X] distutils: 3.12.4
[ ] IPython: not found
[ ] vtk: not found
[ ] enable: not found
[ ] traitsui: not found
[ ] traits: not found
[ ] scikits.samplerate: not found

**Reply by ThomasLecocq** [reply]
don't worry about that, if you have the latest obspy release (1.4.1) and msnoise master, you're good to go!
